src/operations/expansions/offset.js

```diff
@@ -0,0 +1,29 @@
+const flatten = require('../../utils/flatten')
+
+const { geom2, path2 } = require('../../geometry')
+
+const offsetGeom2 = require('./offsetGeom2')
+const offsetPath2 = require('./offsetPath2')
+
+/**
+ * Create offset geometry(s) from the given object(s) using the given options (if any).
+ * @param {Object} options - options for offset
+ * @param {Float} [options.delta=1] - delta of offset (+ to exterior, - from interior)
+ * @param {Integer} [options.segments=0] - segments of rounded ends, or zero for chamfer
+ * @param {Object|Array} objects - object(s) to offset
+ * @return {Object|Array} the offset geometry(s)
+ */
+const offset = (options, ...objects) => {
```

Comment:
ok I confess I am really confused: even going back & forth between files and your screenshots, I do not understand expand vs offset ?

Reply:
Offset 
- path2
- geom2 ; convert to outlines, offset each

Expand
- path2 ; offset on both sides, and close to geom2
- geom2 ; same as offset
- geom3 ; old logic and very very slow

We could do more, but I stopped at this point.

Reply:
sorry for the slow reply on this one: if expand does the same as offset for geom2 it means to me that it is the same operand, so this seems very confusing as it is to be honest
it seems that the main issue here is expand/ offset for path2 since that is the only one that has different results.
I also have a hard time understanding (sorry!) why chamfer is a part of offset and not a different operation ?
I honestly like the results a LOT, but I really think more thought around the API is really worth it for this :)

Reply:
‘Chamfer’ is my term for how to fill the corners, but I think this term makes sense. And we could use the term for rounded rectangles and rounded cubes as well. ‘Segments’ isn’t really the correct term.

The offset() function creates parallel line segments, but that leaves gaps on the corners. The ‘chamfer’ tells the routine how to fill those gaps.

Although, a chamfer() function may be possible I’m not sure it makes sense because the offset() function creates the gaps, and tracks the gaps. Its far more efficient to fill those gaps as part of offset().

Reply:
RESET

Reply:
?? reset

Reply:
I made some changes so let’s start again, with a new discussion.